apps/server/src/app/auth/auth.service.ts

```diff
@@ -0,0 +1,69 @@
+import { BadRequestException, HttpException, Injectable } from "@nestjs/common";
+import { validate } from "class-validator"
+import { compareSync, hashSync } from "bcrypt"
+import { JwtService } from '@nestjs/jwt'
+
+import { UserService, UsersDTO, UserEntity } from "@users/index";
+
+
+class UserAllreadyExistsException extends HttpException {
+}
+
+
+@Injectable()
+export class AuthService {
+
+  constructor(
+    private readonly usersService: UserService,
+    private readonly jwtService: JwtService
+  ) {}
+
+  async login(username: string, password: string): Promise<string> {
+    if (await this.authenticateUser(username, password)) {
+      return this.jwtService.sign({ username })
+    }
+    return null
+  }
+
+  async register(username: string, password: string): Promise<Omit<UserEntity, 'password'>> {
+    if (await this.validateParams(username, password)) {
+      const user = await this.usersService.findOne(username)
+      if (!user) {
+        const result = await this.usersService.save({ username, password: hashSync(password, 10) })
```

Comment:
i think hashSync should be inside the user service, otherwise we could inject the service anywhere and add an plain password

Reply:
nah we can forget this, if i write the password plain i never could login

Reply:
nah we can forget this, if i write the password plain i never could login